pkg/api/types.go

```diff
@@ -705,6 +713,8 @@ type ServiceSpec struct {
 	// not be changed by updates.
 	PortalIP string `json:"portalIP,omitempty"`
 
+	Rewrite bool `json:"rewrite,omitempty" description:"Does the load balancer rewrite packets or pass them through"`
```

Comment:
I'd like to get this change in, but most of my questions were not answered and I still don't understand why we even need to change the Service API at all.

On any given cloud platform is this setting optional, or must it always be set to the same value? Will it ever differ for any 2 services on the same platform? If it must always have the same value,  we should find a different way to pass the value to kube-proxy.

Should Rewrite have any effect for services with createExternalLoadBalancer == false?